workflow/controller/operator.go

```diff
@@ -896,6 +896,15 @@ func (woc *wfOperationCtx) createPVCs() error {
 			*metav1.NewControllerRef(woc.wf, wfv1.SchemaGroupVersionKind),
 		}
 		pvc, err := pvcClient.Create(&pvcTmpl)
+		if err != nil && apierr.IsAlreadyExists(err) {
+			woc.log.Infof("%s pvc has already exists. Workflow is re-using it", pvcTmpl.Name)
+			pvc, err = pvcClient.Get(pvcTmpl.Name, metav1.GetOptions{})
+			if pvc.OwnerReferences == nil {
+				return errors.New(errors.CodeForbidden, "%s pvc has already exists and ownerreference is different")
```

Comment:
Checking `pvc.OwnerReferences  == nil` is not enough. We need to iterate ownership references and verify that our reference is *not*  there.